src/Castle.Core/DynamicProxy/ProxyUtil.cs

```diff
@@ -26,11 +26,73 @@ namespace Castle.DynamicProxy
 #endif
 
 	using Castle.Core.Internal;
+	using Castle.DynamicProxy.Generators;
+	using Castle.DynamicProxy.Internal;
 
 	public static class ProxyUtil
 	{
 		private static readonly SynchronizedDictionary<Assembly, bool> internalsVisibleToDynamicProxy = new SynchronizedDictionary<Assembly, bool>();
 
+		/// <summary>
+		///   Attempts to create a delegate of the specified type to a suitable `Invoke` method
+		///   on the given <paramref name="proxy"/> instance.
+		/// </summary>
+		/// <param name="proxy">The proxy instance to which the delegate should be bound.</param>
+		/// <typeparam name="TDelegate">The type of delegate that should be created.</typeparam>
+		/// <param name="delegate">The produced delegate if the method succeeds; otherwise <see langword="null"/>.</param>
+		/// <returns><see langword="true"/> if the method succeeds; otherwise <see langword="false"/>.</returns>
+		public static bool TryCreateDelegateToMixin<TDelegate>(object proxy, out TDelegate @delegate)
+		{
+			if (TryCreateDelegateToMixin(proxy, typeof(TDelegate), out var d))
+			{
+				@delegate = (TDelegate)(object)d;
+				return true;
+			}
+			else
+			{
+				@delegate = default(TDelegate);
+				return false;
+			}
+		}
+
+		/// <summary>
+		///   Attempts to create a delegate of the specified type to a suitable `Invoke` method
+		///   on the given <paramref name="proxy"/> instance.
+		/// </summary>
+		/// <param name="proxy">The proxy instance to which the delegate should be bound.</param>
+		/// <param name="delegateType">The type of delegate that should be created.</param>
+		/// <param name="delegate">The produced delegate if the method succeeds; otherwise <see langword="null"/>.</param>
+		/// <returns><see langword="true"/> if the method succeeds; otherwise <see langword="false"/>.</returns>
+		public static bool TryCreateDelegateToMixin(object proxy, Type delegateType, out Delegate @delegate)
```

Comment:
Somehow, using the `Try` pattern for this method (and its generic counterpart) feels wrong. In the typical use case of having previously called `options.AddDelegate[Type]Mixin`, there is no reason why users should be expected to deal with failure at this point. *If* this method fails, users will generally have no idea what might have gone wrong, and they likely won't be able to proceed, so it would be better for DynamicProxy to throw an exception directly.